relativedeltafield/__init__.py

```diff
@@ -25,9 +25,31 @@
     r'$'
 )
 
+class iso8601relativedelta(relativedelta):
+	def __init__(self, *args, **kwargs) -> None:
+		if len(args) == 1 and isinstance(args[0], iso8601relativedelta):
+			rd = args[0]
+			self.years = rd.years
+			self.months = rd.months
+			self.days = rd.days
+			self.hours = rd.hours
+			self.minutes = rd.minutes
+			self.seconds = rd.seconds
+			self.microseconds = rd.microseconds
+		else:
+			super().__init__(*args, **kwargs)
+
+	def __str__(self):
+		return format_relativedelta(self)
+
+
 # Parse ISO8601 timespec
-def parse_relativedelta(str):
-	m = iso8601_duration_re.match(str)
+def parse_relativedelta(value):
+	if isinstance(value, relativedelta):
+		return iso8601relativedelta(value)
+	if value is None or value == '':
+		return None
```

Comment:
Why are these needed? The parse function should only be called on strings. See the `to_python` method of the field class, that handles `None` values.